backend/jobs/codeAnalysisJob.ts

```diff
@@ -0,0 +1,370 @@
+  try {
+    const sonarUri = process.env.SONAR_URI;
+    const sonarToken = process.env.SONAR_TOKEN;
+    const projectKey = gitHubOrgName + '_' + repo.name;
+    const metricKeys =
+      'complexity, cognitive_complexity, branch_coverage, coverage, line_coverage, tests, uncovered_conditions, uncovered_lines, test_execution_time, test_errors,  test_failures, test_success_density, skipped_tests, duplicated_blocks, duplicated_files, duplicated_lines, duplicated_lines_density, code_smells, sqale_index, sqale_debt_ratio, sqale_rating, alert_status, quality_gate_details, bugs, reliability_rating, reliability_remediation_effort, vulnerabilities, security_rating, security_remediation_effort, security_hotspots, classes, comment_lines, comment_lines_density, files, lines, ncloc, functions, statements';
+
+    const codeAnalysisResponse = await fetch(
+      `${sonarUri}/api/measures/component?component=${projectKey}&metricKeys=${metricKeys}&additionalFields=metrics`,
+      {
+        method: 'GET',
+        headers: {
+          Authorization: `Basic ${Buffer.from(`${sonarToken}:`).toString('base64')}`,
+        },
+      }
+    );
+
+    if (!codeAnalysisResponse.ok) {
+      throw new Error(
+        `Failed to fetch data from Sonar API: ${codeAnalysisResponse.statusText}`
+      );
+    }
+
+    const responseData = await codeAnalysisResponse.json();
+    const { component, metrics } = responseData;
+
+    if (!component || !component.measures) {
+      throw new Error('Invalid response structure from Sonar API');
+    }
+
+    const metricsArray: string[] = [];
+    const valuesArray: string[] = [];
+    const typesArray: string[] = [];
+    const domainsArray: string[] = [];
+
+    const metricMap = new Map<string, { type: string; domain: string }>();
+
+    metrics.forEach((metric: any) => {
+      metricMap.set(metric.key, {
+        type: metric.type,
+        domain: metric.domain,
+      });
+    });
+
+    component.measures.forEach((measure: any) => {
+      const metricKey = measure.metric;
+      const metricInfo = metricMap.get(metricKey);
+
+      if (metricInfo) {
+        metricsArray.push(metricKey);
+        valuesArray.push(measure.value || '');
+        typesArray.push(metricInfo.type || '');
+        domainsArray.push(metricInfo.domain || '');
+      }
+    });
+
+    const codeAnalysisData = new codeAnalysisDataModel({
+      executionTime: new Date(),
+      gitHubOrgName,
+      teamId: repo.id,
+      repoName: repo.name,
+      metrics: metricsArray,
+      values: valuesArray,
+      types: typesArray,
+      domains: domainsArray,
+    });
+
+    // console.log('Saving code analysis data:', codeAnalysisData);
+
+    await codeAnalysisData.save();
+  } catch (error) {
+    console.error(`Error fetching or saving data for: ${repo.name}`, error);
+  }
+};
+
+export const setupCodeAnalysisJob = () => {
+  // Schedule the job to run every day at midnight
+  cron.schedule('0 0 * * *', async () => {
```

Comment:
Because the githubjob is 0 0 and jirajob is 0 1, maybe this one can set to 0 2, to avoid any complications.